sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -193,6 +193,22 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     Blobs::CreateBlobContainerOptions blobOptions;
     blobOptions.Context = options.Context;
     blobOptions.Metadata = options.Metadata;
+    if (options.AccessType == Models::PublicAccessType::FileSystem)
+    {
+      blobOptions.AccessType = Blobs::Models::PublicAccessType::BlobContainer;
+    }
+    else if (options.AccessType == Models::PublicAccessType::Path)
+    {
+      blobOptions.AccessType = Blobs::Models::PublicAccessType::Blob;
+    }
+    else if (options.AccessType == Models::PublicAccessType::None)
+    {
+      blobOptions.AccessType = Blobs::Models::PublicAccessType::Private;
+    }
+    else
+    {
+      blobOptions.AccessType = Blobs::Models::PublicAccessType(options.AccessType.Get());
```

Comment:
This doesn't make any sense, how can we assume a symbol for datalake also works for blob?

Reply:
We don't, but this is how extensible enums are used right? Customers got something the SDK does not recognize and we just send it to the server to verify. All existing access types are all mapped, while the non-existing ones are sent as-is.
Don't worry, it won't be often for service to add any and if customers decided to use their own strings, we cannot prevent it anyway.